### PR TITLE
fix(component-testing): take screenshot of AUT while hiding other elements

### PR DESCRIPTION
**Pull request by lmiller1990**
It now captures the correct area when calling `cy.screenshot()`. It works the same as the E2E runner: we hide all the other elements, take the screenshot, and make them visible again. This is facilitated via the `before:screenshot` and `after:screenshot` events. It's a little dirty since we must hide some elements inside of `<SplitPane />` that cannot be accessed via `ref`. 

Example screenshot:

image

I added some simple e2e tests. Unit/integration is covered in `server` a little, but primarily `driver`. The tests I added compare the dimensions as well as the raw contents (as a `Buffer` via `fs.readFile`) of the screenshot vs a pre-existing one that I determined to be correct.

There are still some problems with `cy.screenshot` in general, but this covers the basic use cases. The remaining issues are related to the AUT scaling properly. [Documented here](https://cypress-io.atlassian.net/browse/CT-270).

### CODE REVIEW

File: packages/runner-ct/cypress/plugins/index.js [hunk 1]
```diff
@@ -22,6 +27,41 @@ function injectStylesInlineForPercyInPlace (webpackConfig) {
  */
 module.exports = (on, config) => {
   on('task', percyHealthCheck)
+  on('task', {
+    compareImages (screenshotName) {
+      const readAsync = (p) => {
+        return new Promise((res, rej) => {
```
Comment (dmtrKovalenko) [inline packages/runner-ct/cypress/plugins/index.js:33]: We can use more stable thing to compare images https://github.com/dmtrKovalenko/odiff or at least pixelmatch.

Comment (lmiller1990) [reply packages/runner-ct/cypress/plugins/index.js:35]: Neat, I will update the tests to use this.

**Comment by lmiller1990** [comment]
Hmm, odiff error: 

```
From Node.js Internals:
    Error: /root/cypress/node_modules/odiff-bin/3_______________________________________________/i/odiff-9cfb2f81/bin/ODiffBin: error while loading shared libraries: libjpeg.so.8: cannot open shared object file: No such file or directory
```

**Comment by dmtrKovalenko** [comment]
I'll take a look on this :)

**Comment by dmtrKovalenko** [comment]
@lmiller1990 It looks like an ubuntu v18-specific issue, I didn't expect such errors that's nice that we found it! 

But I realized that odiff or library are not required here for us because we can use Percy itself – I updated the code to use percy for screenshot testing. 

> And also I moved screenshot logic in a separate hook cause this file is becoming large and it looks like adding more functions to the root component can harm readability. Please do not be offended by unexpected push :)

File: packages/runner-ct/cypress/component/screenshot.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+describe('screenshot', () => {
+  // beforeEach(() => {
+  //   cy.task('clearScreenshots')
+  // })
+
+  // afterEach(() => {
+  //   cy.task('clearScreenshots')
+  // })
+
```
Comment (dmtrKovalenko) [inline packages/runner-ct/cypress/component/screenshot.spec.tsx:46]:
```suggestion
```

File: packages/runner-ct/src/app/RunnerCt.scss [hunk 3]
```diff
@@ -12,6 +12,10 @@ main.app-ct {
   font-size: $font-size;
 }
 
+.d-none {
```
Comment (dmtrKovalenko) [inline packages/runner-ct/src/app/RunnerCt.scss:15]: IMHO this name can be unclear when reading it from the code, maybe we can create a more descriptive name? For reference: tailwind css using `hidden` for this [source](https://tailwindcss.com/docs/display)

Comment (lmiller1990) [reply packages/runner-ct/src/app/RunnerCt.scss:15]: How about `display-none`? 🤔

Comment (lmiller1990) [reply packages/runner-ct/src/app/RunnerCt.scss]: Changed to display-none... I think this is much more clear than hidden - it's literally the exact css, `display: none`. 

We can revisit these at some point, we should probably grab a library that has these to save us the effort of discussing class names (eg tailwind, but probably something a little smaller).

**Review by dmtrKovalenko — APPROVED** [review]

**Pull request merged**